api/bots/gcalendar/gcalendar.py

```diff
@@ -0,0 +1,349 @@
+import httplib2
+import logging
+import os
+import sys
+
+# Uses Google's Client Library
+#   pip install --upgrade google-api-python-client
+from apiclient import discovery, errors
+from oauth2client.client import HttpAccessTokenRefreshError
+from oauth2client.file import Storage
+
+# pip install --upgrade dateparser
+import dateparser
+
+from pytz.exceptions import UnknownTimeZoneError
+from tzlocal import get_localzone
+
+# Avoid cache-related warnings
+# https://github.com/google/google-api-python-client/issues/299
+logging.getLogger('googleapiclient.discovery_cache').setLevel(logging.ERROR)
+
+# Modify this to match where your credentials are stored.
+# It should be the same path you used in the
+# '--credential_path' argument, when running the script
+# in "api/integrations/google/oauth.py".
+#
+# $ python api/integrations/google/oauth.py \
+#   --secret_path <your_client_secret_file_path> \
+#   --credential_path <your_credential_path> \
+#   -s https://www.googleapis.com/auth/calendar
+#
+# If you didn't specify any, it should be in the default
+# path (~/.google_credentials.json)
+CREDENTIAL_PATH = '~/.google_credentials.json'
+
+class UnknownMessageTypeError(Exception):
+    def __init__(self, msgtype):
+        self.type = msgtype
+        self.message = 'Unknown message type {type}'.format(type=msgtype)
+
+    def __str__(self):
+        return self.message
+
+class MessageParseError(Exception):
+    def __init__(self, error_id):
+        self.error_id = error_id
+
+        details = ''
+
+        if error_id == 1:
+            details = 'Unknown message format'
+        elif error_id == 2:
+            details = 'Unknown date format'
+        elif error_id == 3:
+            details = 'The specified timezone doesn\'t exist'
+
+        self.message = details
+
+    def __str__(self):
+        return self.message
+
+def reply_message(msg, original_msg, client):
```

Comment:
This function can now be substituted by calls to bothandler.send_reply(original_msg, msg) ?
That allows the related Exception to be removed.